static/js/services/select2-search.js

```diff
@@ -155,8 +155,27 @@ angular.module('inboxServices').factory('Select2Search',
             });
           selectEl.after(button);
         }
+
+        // Hydrate and re-set real doc on change
+        // !tags -> only support single values, until there is a use-case
+        if (!tags) {
+          selectEl.on('select2:select', function(e) {
+            var docId = e.params &&
+                        e.params.data &&
+                        e.params.data.id;
+
+            if (docId) {
+              DB().get(docId).then(function(doc) {
+                selectEl.select2('data')[0].doc = doc;
+
+                selectEl.trigger('change');
+              });
+            }
+          });
+        }
       };
 
+
       initSelect2(selectEl);
       return resolveInitialValue(selectEl, initialValue);
```

Comment:
Setting the initial value should hydrate the doc. I don't know the easiest way to do that, maybe just explicitly call `DB.get`?